### DISCUSSION TITLE
Config WSL with [desktop](https://github.com/desktop/desktop)?

### DISCUSSION THREAD

**Discussion by timsu92** [git-ecosystem/git-credential-manager#787] [Ideas]
In the [docs](https://github.com/GitCredentialManager/git-credential-manager/blob/main/docs/wsl.md#configuring-wsl-without-git-for-windows), it says I can config WSL without Git for Windows. It asks me to install [latest GCM](https://aka.ms/gcm/latest) first, and then config Git to it.
However, there's a `git-credential-manager-core.exe` in GitHub Desktop's installation location as well. **I wonder if we can use it directly instead of installing a new software?**

Currently, I have GitHub Desktop 3.0.3 comes with GCM 2.0.632 installed on Windows 11. In order to check file consistency, I downloaded [the zip file of GCM 2.0.632](https://github.com/GitCredentialManager/git-credential-manager/releases/download/v2.0.632/gcmcore-win-x86-2.0.632.34631.zip) and compared `C:\Users\TimSu\AppData\Local\GitHubDesktop\app-3.0.3\resources\app\git\mingw64\libexec\git-core` with `<the zip file>\gcmcore-win-x86-2.0.632.34631`, and here are their differences:
| file name | file in Desktop | file in GCM standalone |
| --------- | --------------- | ---------------------- |
| `Atlassian.Bitbucket.UI.Shared.dll` | Yes, larger | Yes, smaller |
| `git-bisect` | Yes | No |
| `git-filter-branch` | Yes | No |
| `GitHub.UI.Shared.dll` | Yes, larger | Yes, smaller |
| 67 files start with `git-lfs` | Yes | No |
| 5 files start with `git-merge` | Yes | No |
| `mergetools` | Yes | No |
| `git-quiltimport` | Yes | No |
| `git-request-pull` | Yes | No |
| `git-sh-i18n` | Yes | No |
| `git-sh-setup` | Yes | No |
| `git-submodule` | Yes | No |
| `git-subtree` | Yes | No |
| `git-update` | Yes | No |
| `git-web--browse` | Yes | No |
| `validatesign.timestamp` | No | Yes |
| `WebView2Loader.dll` | No | Yes |

As I'm a newbie to Git, I don't know what do those file differences mean. But if it's feasible to use GCM brought by Desktop, it's gonna be great since Desktop checks for updates every time it starts up, but GCM don't.
However, doing this requires a helper, I guess, because the file structure isn't like that is of GCM. It seems that Desktop put different versions of it into different folders, while GCM just replaces old files.

What do you think? Do you think this is something should be put on to-do list?

**Comment by Guillermo250** [comment]
I need help mi phonecell are controlled for somebody. I dont know what i can do it

**Reply by timsu92** [reply]
No, I can't, if the translator worked as expected.
Please note that this is a international website. It's better to write a least in English, so that others know what you are saying

**Comment by becm** [comment]
The `git` distribution in [Github Desktop](https://desktop.github.com/download/) and [Atlassian Sourcetree](https://www.atlassian.com/software/sourcetree) should just be a [compact version](https://github.com/git-for-windows/git/wiki/MinGit) of `git-for-windows`.
It _should_ include a fully functional distribution of Git Credential Manager.

You will have to re-target the WSL setting each time the respective Windows software receives an update.
The location of the respective binaries (…/GithubDesktop/**app-3.0.3**/resources/app/git/…) change in each version.
